src/core/components/mod-icon/mod-icon.ts

```diff
@@ -146,11 +156,11 @@ export class CoreModIconComponent implements OnInit, OnChanges {
             return;
         }
 
-        this.iconUrl = CoreTextUtils.decodeHTMLEntities(this.iconUrl);
+        this.iconUrl.update(value => CoreTextUtils.decodeHTMLEntities(value));
 
         // If it's an Moodle Theme icon, check if filtericon is set and use it.
-        if (this.iconUrl && CoreUrlUtils.isThemeImageUrl(this.iconUrl)) {
-            const filter = CoreUrlUtils.getThemeImageUrlParam(this.iconUrl, 'filtericon');
+        if (this.iconUrl && CoreUrlUtils.isThemeImageUrl(this.iconUrl())) {
```

Comment:
```suggestion
        if (CoreUrlUtils.isThemeImageUrl(this.iconUrl())) {
```
I think this part can be removed, I guess it was added at some point to avoid supplying an undefined value to isThemeImageUrl but I think this can no longer happen.